File: data_structures/JavaScript/tree.js
//Create a tree with pseudoclassical instantiation
var Tree = function(value) {
  this.val = value;
  this.children = [];
}

//Each node is a tree, so we create a new tree node and
//add it to the current node's children
Tree.prototype.addChild = function(val) {
  this.children.push(new Tree(val));
}

//Depth first search to see if a tree contains a value
Tree.prototype.contains = function(val) {
  if (this.val === val) {
    return true;
  } else {
    for (var i = 0; i < this.children.length; i++) {
      if (this.children[i].contains(val)) {
        return true;
      }
    }
    return false;
  }

  return false;
}

File: data_structures/Python/tree.py
#Implement a Tree class
class Tree():

    def __init__(self, value):
        self.val = value
        self.children = []

    def addChild(self, value):
        self.children.append(Tree(value))

    #Contains is a depth first search
    def contains(self, value):
        if self.val == value:
            return True
        else:
            for node in self.children:
                if node.contains(value):
                    return True
            return False



# tree = Tree("bob")
# tree.addChild("tim")
# tree.addChild("larry")
# print(tree.contains("bob"))
# print(tree.contains("tim"))
# print(tree.contains("larry"))
# print(tree.contains("jim"))